test: add alert system E2E tests

vinistevam opened the pull request:
<!--
Please submit this PR as a draft initially.
Do not mark it as "Ready for review" until the template has been completely filled out, and PR status checks have passed at least once.
-->

## **Description**

This PR adds new tests covering the alert system.
- Test general banner blockaid alert (Security Alerts API)
- Test inline alert SIWE origin mismatch alert

<!--
Write a short description of the changes included in this pull request, also include relevant motivation and context. Have in mind the following questions:
1. What is the reason for the change?
2. What is the improvement/solution?
-->

## **Related issues**

Fixes: https://github.com/MetaMask/mobile-planning/issues/2160

## **Manual testing steps**

1. E2E Bitrise

## **Screenshots/Recordings**

<!-- If applicable, add screenshots and/or recordings to visualize the before and after of your change. -->

### **Before**

<!-- [screenshots/recordings] -->

### **After**

<!-- [screenshots/recordings] -->

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Mobile Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

vinistevam committed:
feat: add initial structure for Alert System

vinistevam committed:
fix lint and add initial structure

vinistevam committed:
add unit tests

vinistevam committed:
add use memo

vinistevam committed:
fix sonar issue

vinistevam committed:
Merge branch 'main' into feat/initial-structure-alert-system

vinistevam committed:
feat: add AlertModal component

vinistevam committed:
Merge branch 'main' into feat/initial-structure-alert-system

vinistevam committed:
unit test and clean up

vinistevam committed:
changed actions to action alert property

vinistevam committed:
add unit test and applied suggestions

vinistevam committed:
Add unit tests

vinistevam committed:
Merge branch 'feat/initial-structure-alert-system' into feat/add-alert-modal

vinistevam committed:
add unit test

vinistevam committed:
Merge branch 'feat/add-alert-modal' of https://github.com/MetaMask/metamask-mobile into feat/add-alert-modal

vinistevam committed:
Applied suggestions

vinistevam committed:
Merge branch 'main' into feat/initial-structure-alert-system

vinistevam committed:
Merge branch 'feat/initial-structure-alert-system' into feat/add-alert-modal

vinistevam committed:
fix unit tests

vinistevam committed:
fix lint

vinistevam committed:
Merge branch 'feat/initial-structure-alert-system' into feat/add-alert-modal

vinistevam committed:
feat: add Confirm Alert Modal component

vinistevam committed:
feat: add Confirm Alert Modal component

vinistevam committed:
fix unit test

vinistevam committed:
fix unit test

vinistevam committed:
fix sonar issues

vinistevam committed:
fix sonar issues

vinistevam committed:
feat: add GeneralAlertBanner component

vinistevam committed:
Merge branch 'main' into feat/add-alert-modal

vinistevam committed:
Merge branch 'feat/add-alert-confirm' into feat/add-general-alert-banner

vinistevam committed:
feat: add MultipleAlertModal component

vinistevam committed:
Merge branch 'main' into feat/add-alert-modal

vinistevam committed:
Merge branch 'feat/add-alert-modal' into feat/add-alert-confirm

vinistevam committed:
fix: applied suggestions

vinistevam committed:
Merge branch 'feat/add-alert-confirm' into feat/add-general-alert-banner

vinistevam committed:
fix alertKey state unit tests

vinistevam committed:
Merge branch 'feat/add-general-alert-banner' into feat/add-multiple-alert-modal

vinistevam committed:
feat: add inline alert component

vinistevam committed:
Merge branch 'main' into feat/add-alert-modal

vinistevam committed:
Merge branch 'feat/add-alert-modal' into feat/add-alert-confirm

vinistevam committed:
Merge branch 'feat/add-alert-confirm' into feat/add-general-alert-banner

vinistevam committed:
Merge branch 'feat/add-general-alert-banner' into feat/add-multiple-alert-modal

vinistevam committed:
fix: change default return for colors

vinistevam committed:
fix lint

vinistevam committed:
Merge branch 'main' into feat/add-alert-confirm

vinistevam committed:
Merge branch 'feat/add-alert-confirm' into feat/add-general-alert-banner

vinistevam committed:
Merge branch 'feat/add-general-alert-banner' into feat/add-multiple-alert-modal

vinistevam committed:
applied suggestions

vinistevam committed:
Merge branch 'feat/add-general-alert-banner' into feat/add-multiple-alert-modal

vinistevam committed:
fix: apply suggestions

vinistevam committed:
Merge branch 'feat/add-multiple-alert-modal' into feat/add-inline-alert

vinistevam committed:
Merge branch 'main' into feat/add-alert-confirm

vinistevam committed:
Merge branch 'feat/add-alert-confirm' into feat/add-general-alert-banner

vinistevam committed:
Merge branch 'feat/add-general-alert-banner' into feat/add-multiple-alert-modal

vinistevam committed:
Merge branch 'feat/add-multiple-alert-modal' into feat/add-inline-alert

vinistevam committed:
Merge branch 'main' into feat/add-general-alert-banner

vinistevam committed:
Merge branch 'feat/add-general-alert-banner' into feat/add-multiple-alert-modal

vinistevam committed:
Merge branch 'feat/add-multiple-alert-modal' into feat/add-inline-alert

vinistevam committed:
feat: add blockaid banner alert

vinistevam committed:
fix: add locales

vinistevam committed:
add constants

vinistevam committed:
Merge branch 'main' into feat/add-general-alert-banner

vinistevam committed:
Merge branch 'feat/add-general-alert-banner' into feat/add-multiple-alert-modal

vinistevam committed:
Merge branch 'feat/add-multiple-alert-modal' into feat/add-inline-alert

vinistevam committed:
fix: add unit tests

vinistevam committed:
Merge branch 'feat/add-inline-alert' into feat/add-blockaid-alert

vinistevam committed:
Merge branch 'main' into feat/add-general-alert-banner

vinistevam committed:
Merge branch 'feat/add-general-alert-banner' into feat/add-multiple-alert-modal

vinistevam committed:
Merge branch 'feat/add-multiple-alert-modal' into feat/add-inline-alert

vinistevam committed:
Merge branch 'feat/add-inline-alert' into feat/add-blockaid-alert

vinistevam committed:
feat: integrate inline alert into the Alert System

vinistevam committed:
fix: unit tests

vinistevam committed:
Merge branch 'main' into feat/add-multiple-alert-modal

vinistevam committed:
Merge branch 'feat/add-multiple-alert-modal' into feat/add-inline-alert

vinistevam committed:
Merge branch 'feat/add-inline-alert' into feat/add-blockaid-alert

vinistevam committed:
fix: applied suggestions

vinistevam committed:
fix: pipeline

vinistevam committed:
fix: add unit tests

vinistevam committed:
fix: fix e2e adapting to the new alert system

vinistevam committed:
fix: checkbox from UX feedback

vinistevam committed:
fix: apply UX feedback

vinistevam committed:
Merge branch 'main' into feat/add-inline-alert

vinistevam committed:
add: inline alert component

vinistevam committed:
fix: sort imports

vinistevam committed:
Merge branch 'feat/add-inline-alert' into feat/add-blockaid-alert

vinistevam committed:
fix: new line

vinistevam committed:
Merge branch 'main' into feat/add-blockaid-alert

vinistevam committed:
Merge branch 'main' into feat/add-blockaid-alert

vinistevam committed:
fix: applied suggestions

vinistevam committed:
Merge branch 'main' into feat/add-blockaid-alert

vinistevam committed:
fix: domain without protocol

vinistevam committed:
test: add alert system e2e

vinistevam committed:
fix: checkbox test ids

vinistevam committed:
Merge branch 'main' into feat/add-blockaid-alert

vinistevam committed:
Merge branch 'feat/add-blockaid-alert' into feat/add-siwe-alert-request-from

vinistevam committed:
fix: value styles in infoRow

digiwand commented on e2e/specs/confirmations-redesigned/signatures/alert-system.spec.js:
```diff
@@ -0,0 +1,187 @@
+
+describe(SmokeConfirmationsRedesigned('Alert System - Signature'), () => {
+  beforeAll(async () => {
+    jest.setTimeout(2500000);
```
hm this timeout is 2500000 ms → 41.67 min. Is there a way around this? as in do we need the timeout?

vinistevam replied:
Good point. I've removed it and let the default timeout handle it.

vinistevam committed:
fix: applied suggestions

vinistevam committed:
Merge branch 'main' into feat/add-siwe-alert-request-from

vinistevam committed:
Merge branch 'feat/add-siwe-alert-request-from' into test/add-e2e-alert-system

vinistevam committed:
fix: merge conflicts

vinistevam committed:
Merge branch 'feat/add-siwe-alert-request-from' into test/add-e2e-alert-system

vinistevam committed:
Merge branch 'main' into feat/add-siwe-alert-request-from

vinistevam committed:
fix: apply suggestions

vinistevam committed:
Merge branch 'main' into feat/add-siwe-alert-request-from

vinistevam committed:
Merge branch 'feat/add-siwe-alert-request-from' into test/add-e2e-alert-system

vinistevam committed:
Merge branch 'main' into test/add-e2e-alert-system

vinistevam committed:
fix: merge conflicts

jpuri commented on app/components/Views/confirmations/components/UI/InfoRow/InfoRow.tsx:
```diff
@@ -38,7 +38,7 @@ const InfoRow = ({
       {Boolean(label) && (
         <View style={styles.labelContainer}>
-          <Text variant={TextVariant.BodyMDMedium} color={variant}>{label}</Text>
+          <Text variant={TextVariant.BodyMDMedium} color={variant} >{label}</Text>
```
This format looks wrong.

vinistevam replied:
True, I've fixed [here](https://github.com/MetaMask/metamask-mobile/pull/14162/commits/fcc87680d23338e6ecc95f6d0b15349348422fe3) in the following PR. I don't know why, but I have to constantly run `yarn lint:fix`.

jpuri reviewed (dismissed):
Great work 👍

pedronfigueiredo reviewed (dismissed):
LGTM

vinistevam committed:
Merge branch 'main' into test/add-e2e-alert-system

vinistevam committed:
feat: add metrics for Alert System (#14162)

<!--
Please submit this PR as a draft initially.
Do not mark it as "Ready for review" until the template has been
completely filled out, and PR status checks have passed at least once.
-->

## **Description**
This PR aims to implement metrics to the alert system and use them in
signature requests that use the alert system. It's not included in
transactions in this PR.

```
alert_triggered_count: total number of alerts triggered within that confirmation
alert_triggered (list): name of the alerts displayed, e.g. insufficient_funds_for_gas, pending_transaction
alert_visualized_count: total number of alerts visualized by the user within that confirmation
alert_visualized (list): name of the alerts that the user visualized regardless if by clicking on the key or going through it after clicking confirm
alert_resolved_count (list): total number of alerts that were triggered but then resolved before the user confirmed or rejected the request
alert_resolved (list): name of alerts that were triggered but then resolved before the user confirmed
```


<!--
Write a short description of the changes included in this pull request,
also include relevant motivation and context. Have in mind the following
questions:
1. What is the reason for the change?
2. What is the improvement/solution?
-->

## **Related issues**

Fixes: https://github.com/MetaMask/MetaMask-planning/issues/3923

## **Manual testing steps**

**Confim case**
1. Go to this the test dapp
2. click on `Sign In With Ethereum (Bad Domain)`
3. Click on the inline alert
4. Close
5. Click on `Review alert`
6. Acknowledge the alert
7. Acknowledge the Friction modal and confirm the request

**Reject case**
1. Go to this the test dapp
2. click on `Sign In With Ethereum (Bad Domain)`
3. Click on the inline alert
6. Acknowledge the alert
7. Cancel the request

**Blockaid Alert**
1. Go to this the test dapp
2. click on `Malicious Permit` in the Signature and Blockaid section
4. Reject or Confirm the request

Dev env we can see the events via the terminal in the build we have to
go to Segment to check the events.

## **Screenshots/Recordings**


[alert_metrics.webm](https://github.com/user-attachments/assets/59a8b208-520b-4248-8976-9d280c00a96f)

<!-- If applicable, add screenshots and/or recordings to visualize the
before and after of your change. -->

### **Before**

<!-- [screenshots/recordings] -->

### **After**

<!-- [screenshots/recordings] -->

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Contributor
Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Mobile
Coding
Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format
if applicable
- [x] I’ve applied the right labels on the PR (see [labeling
guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)).
Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the
app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described
in the ticket it closes and includes the necessary testing evidence such
as recordings and or screenshots.

vinistevam committed:
Revert "feat: add metrics for Alert System" (#14245)

Reverts MetaMask/metamask-mobile#14162

vinistevam committed:
Merge branch 'main' into test/add-e2e-alert-system
